cpp/translation_db_tool.cc

```diff
@@ -0,0 +1,118 @@
+/** \file translation_db_tool.cc
+ *  \brief A tool for reading/editing of the translation SQL table.
+ *  \author Dr. Johannes Ruscheinski (johannes.ruscheinski@uni-tuebingen.de)
+ *
+ *  \copyright 2015 Universitätsbiblothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <iostream>
+#include <map>
+#include <cstdlib>
+#include <cstring>
+#include "DbConnection.h"
+#include "DbResultSet.h"
+#include "DbRow.h"
+#include "IniFile.h"
+#include "SimpleXmlParser.h"
+#include "StringUtil.h"
+#include "util.h"
+
+
+void Usage() {
+    std::cerr << "Usage: " << progname << " command [args]\n\n";
+    std::cerr << "       Possible commands are:\n";
+    std::cerr << "       get_next start_index language_code\n";
+    std::cerr << "       insert index language_code text\n";
+    std::exit(EXIT_FAILURE);
+}
+
+
+void GetNext(DbConnection * const connection, const unsigned start_index, const std::string &language_code) {
+    const std::string SELECT_STMT("SELECT * FROM translations WHERE index > " + std::to_string(start_index));
```

Comment:
SELECT `id` FROM `translations` WHERE id NOT IN (SELECT id FROM `translations` WHERE `language_code` = "fra") LIMIT 1;